rust/worker/src/execution/operators/filter.rs

```diff
@@ -0,0 +1,519 @@
+use std::{
```

Comment:
Can we add some basic unit tests here? 

The python property tests will ofc be a bulwark but we should have some sanity in rust for now at least.

Reply:
The unit tests will be added after I move the test utilities into respective crates.

Reply:
Update: I've merged a separate PR that moves test utilities around. I'll add the tests in the middle of the stack where the `GetOrchestrator` is implemented, and that PR will serve as a checkpoint that is safe to merge to main.

Reply:
Update: Unit tests have been introduced. Checkout https://github.com/chroma-core/chroma/pull/2985